src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,78 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkIsUserNull(user);
+        checkLoginForNull(user);
+        checkPasswordForNull(user);
+        checkLoginsLength(user);
+        checkPasswordsLength(user);
+        checkAge(user);
+        checkUserExistence(user);
+        storageDao.add(user);
+        return storageDao.get(user.getLogin());
+    }
+
+    private void checkUserExistence(User user) {
+        if (isUserExist(user)) {
+            throw new RegistrationException("The storage already have user with login "
+                    + user.getLogin());
+        }
+    }
+
+    private boolean isUserExist(User user) {
+        return storageDao.get(user.getLogin()) != null;
+    }
+
+    private static void checkAge(User user) {
+        int usersAge = user.getAge();
+        if (usersAge < MIN_AGE) {
+            throw new RegistrationException("User's is not valid! It's " + usersAge + ", but min "
+                    + "age is " + MIN_AGE);
+        }
+    }
+
+    private static void checkPasswordsLength(User user) {
```

Comment:
be careful with method/parameter/class names. Typo or wrong name can affect on production 
```suggestion
    private static void checkPasswordLength(User user) {
```